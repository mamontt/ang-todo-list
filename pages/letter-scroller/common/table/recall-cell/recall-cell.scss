@import '~styles/constants/index';

.recall-cell {
    align-items: center;
    display: flex;
    height: 100%;
    justify-content: center;
    width: 100%;

    &__icon {
        color: $icon-grey-default;
        margin: 0 12px;
    }
}

