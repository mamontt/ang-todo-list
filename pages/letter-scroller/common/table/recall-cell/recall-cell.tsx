/*
 * VTB Group. Do not reproduce without permission in writing.
 * Copyright (c) 2019 VTB Group. All rights reserved.
 */
import * as React from 'react';
import {get, includes} from 'lodash';
import {Icons} from '@vtb/fe-ui-icon';
import styleNames from '@vtb/services/style-names';
import {translate} from './../../../../../utils/translate';
import styles from './recall-cell.scss';
import {STATUSES} from '../../../../../constants/statuses';

const sn = styleNames(styles);

export const hiddenRecallIconStatuses = [
    STATUSES.DRAFT,
    STATUSES.REJECTED,
    STATUSES.WRONG_E_SIGNATURE,
    STATUSES.DETAILS_ERROR,
    STATUSES.EXECUTED,
    STATUSES.DELETED,
    STATUSES.NOT_RECEIVED_BY_ABS
];

export function RecallCell({row}: {row?: {cancelReq?: {baseStatus?: string}}}) {
    const cancelBaseStatusName = get(row, 'cancelReq.baseStatusName', translate('common.recall'));
    const cancelBaseStatus = get(row, 'cancelReq.baseStatus', null);
    const showIcon = row && row.cancelReq && ['NEW', 'SIGNED'].includes(row.cancelReq.baseStatus) &&
        !includes(hiddenRecallIconStatuses, cancelBaseStatus);
    return showIcon
        ? (
            <div className={sn('recall-cell')}>
                <div className={sn('recall-cell__icon')}>
                    <Icons.Recall alt={cancelBaseStatusName} />
                </div>
            </div>
        )
        : null;
}
