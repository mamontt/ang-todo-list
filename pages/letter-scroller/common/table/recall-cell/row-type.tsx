/*
 * VTB Group. Do not reproduce without permission in writing.
 * Copyright (c) 2019 VTB Group. All rights reserved.
 */
import {Branch, Client} from '../../../../../common-types';
import {AttachmentType} from '../../../../../modules/attachments/attachment';

export type Row = {
    attachments: Array<AttachmentType>;
    bankEmployeeNote: string;
    branchSnapshot: Branch;
    clientSnapshot: Client;
    createDate: string;
    documentDate: string;
    documentNumber: string;
    edocRefId: number;
    favourite: boolean;
    hasRelatedDocuments: boolean;
    id: number;
    letterType: string;
    read: boolean;
    status: {
        id: string;
    };
    toBank: boolean;
    topic: string;
    row: RowItem;
}

export type RowItem = {
    attachments: Array<AttachmentType>;
    hasRelatedDocuments: boolean;
    cancelReq: {
        baseStatus: string;
    };
}
