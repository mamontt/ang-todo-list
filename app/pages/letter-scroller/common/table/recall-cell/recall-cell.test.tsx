/*
 * VTB Group. Do not reproduce without permission in writing.
 * Copyright (c) 2019 VTB Group. All rights reserved.
 */
import * as React from 'react';
import {shallow} from 'enzyme';
import {RecallCell} from './recall-cell';

jest.mock('./../../../../../utils/translate', () => ({
    translate: (text: string) => text
}));

describe('RecallCell', () => {
    it('render RecallCell', () => {
        const wrapper = shallow(<RecallCell />);
        expect(wrapper).toMatchSnapshot();
    });
    it('render RecallCell with empty row', () => {
        const wrapper = shallow(<RecallCell row={{}} />);
        expect(wrapper).toMatchSnapshot();
    });
    it('render RecallCell with empty cancelReq', () => {
        const wrapper = shallow(<RecallCell row={{cancelReq: {}}} />);
        expect(wrapper).toMatchSnapshot();
    });
    it('render RecallCell with new status', () => {
        const wrapper = shallow(<RecallCell row={{cancelReq: {baseStatus: 'NEW'}}} />);
        expect(wrapper).toMatchSnapshot();
    });
    it('render RecallCell with executed status', () => {
        const wrapper = shallow(<RecallCell row={{cancelReq: {baseStatus: 'EXECUTED'}}} />);
        expect(wrapper).toMatchSnapshot();
    });
});

