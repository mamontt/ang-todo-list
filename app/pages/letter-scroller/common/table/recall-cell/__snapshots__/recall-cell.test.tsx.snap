// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecallCell render RecallCell 1`] = `""`;

exports[`RecallCell render RecallCell with empty cancelReq 1`] = `""`;

exports[`RecallCell render RecallCell with empty row 1`] = `""`;

exports[`RecallCell render RecallCell with executed status 1`] = `""`;

exports[`RecallCell render RecallCell with new status 1`] = `
<div
  className="recall-cell"
>
  <div
    className="recall-cell__icon"
  >
    <IconComponent
      alt="common.recall"
    />
  </div>
</div>
`;
